packages/block-library/src/embed/embed-bottom-sheet.native.js

```diff
@@ -56,6 +63,17 @@ const EmbedBottomSheet = ( { value, label, isVisible, onClose, onSubmit } ) => {
 		}
 	}, [ url, onSubmit, value ] );
 
+	/**
+	 * If the Embed Bottom Sheet component does not utilize a bottom sheet then the onDismiss action is not
+	 * called. Here we are wiring the onDismiss to the onClose callback that gets triggered when input is submitted.
+	 */
+	const performOnCloseOperations = () => {
```

Comment:
The idea here is that when the Link Picker is being used the `onDimiss` is not called. Therefore when `onClose` is called, it functions as `onDismiss` as well.